pkg/orchestrator/provider/azure/scanestimation/pricefetcher.go

```diff
@@ -0,0 +1,219 @@
+	symbol := findClosestSSDSizeSymbol(diskSize)
+	odataFilter = fmt.Sprintf(odataFilter, region, symbol, symbol)
+
+	return o.getRetailPrice(ctx, odataFilter)
+}
+
+func (o *PriceFetcherImpl) GetDataTransferPerGBCost(ctx context.Context, destRegion string) (float64, error) {
+	odataFilter := fmt.Sprintf("armRegionName eq '%s' and contains(meterName,'Region Data Transfer')", destRegion)
+
+	return o.getRetailPrice(ctx, odataFilter)
+}
+
+func (o *PriceFetcherImpl) GetInstancePerHourCost(ctx context.Context, region string, vmSize armcompute.VirtualMachineSizeTypes, marketOption MarketOption) (float64, error) {
+	var odataFilter string
+
+	if marketOption == MarketOptionSpot {
+		odataFilter = spotVirtualMachineSizeToFilters[vmSize]
+		odataFilter = fmt.Sprintf(odataFilter, region)
+	} else {
+		odataFilter = virtualMachineSizeToFilters[vmSize]
+		odataFilter = fmt.Sprintf(odataFilter, region)
+	}
+
+	return o.getRetailPrice(ctx, odataFilter)
+}
+
+func (o *PriceFetcherImpl) GetBlobStoragePerGBCost(ctx context.Context, region string, storageAccountType armcompute.StorageAccountTypes) (float64, error) {
+	odataFilter := blobStorageAccountToFilter[storageAccountType]
+	odataFilter = fmt.Sprintf(odataFilter, region)
+
+	return o.getRetailPrice(ctx, odataFilter)
+}
+
+func (o *PriceFetcherImpl) getRetailPrice(ctx context.Context, odataFilter string) (float64, error) {
+	urlWithFilter, err := addQueryParamToURL(priceListBaseURL, "$filter", odataFilter)
+	if err != nil {
+		return 0, fmt.Errorf("failed to add query params to url: %w", err)
+	}
+
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, urlWithFilter, nil)
+	if err != nil {
+		return 0, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	resp, err := o.client.Do(request)
+	if err != nil {
+		return 0, fmt.Errorf("failed to send request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return 0, fmt.Errorf("status code is not 200 when calling url: %v. status code: %v", urlWithFilter, resp.StatusCode)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return 0, fmt.Errorf("failed to read body: %w", err)
+	}
+
+	var data Data
+
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		return 0, fmt.Errorf("failed to unmarshal JSON: %w", err)
+	}
+
+	if data.Count != 1 {
+		return 0, fmt.Errorf("found %v items from response, excpecting 1. response: %s", data.Count, body)
+	}
+
+	return data.Items[0].RetailPrice, nil
+}
+
+func addQueryParamToURL(baseURL, key, value string) (string, error) {
+	u, err := url.Parse(baseURL)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse url %v: %w", baseURL, err)
+	}
+
+	encodedParam := fmt.Sprintf("%s=%s", key, url.QueryEscape(value))
+
+	// If there are already parameters, append the new one with '&', otherwise just add the new one
+	if u.RawQuery != "" {
+		u.RawQuery = u.RawQuery + "&" + encodedParam
+	} else {
+		u.RawQuery = encodedParam
+	}
```

Comment:
I used u.Query initially , but then I noticed that since the returned query values represented in a map, there is no way for me to determine the order in which they reassemble. That broke the api since it moved the "api-version=2023-01-01-preview" param to the end.